tests/boards/intel_adsp/smoke/src/cpus.c

```diff
@@ -84,19 +84,19 @@ static void core_smoke(void *arg)
 	zassert_equal(cpu, arch_curr_cpu()->id, "wrong cpu");
 
 	/* Un/cached regions should be configured and distinct */
-	zassert_equal(&tag, arch_xtensa_cached_ptr((void *)&tag),
+	zassert_equal(&tag, sys_cache_cached_ptr_get((void *)&tag),
 		      "stack memory not cached");
-	zassert_not_equal(&tag, arch_xtensa_uncached_ptr((void *)&tag),
+	zassert_not_equal(&tag, sys_cache_uncached_ptr_get((void *)&tag),
 			  "stack memory not cached");
-	zassert_not_equal(&static_tag, arch_xtensa_cached_ptr((void *)&static_tag),
+	zassert_not_equal(&static_tag, sys_cache_cached_ptr_get((void *)&static_tag),
 		      "stack memory not cached");
-	zassert_equal(&static_tag, arch_xtensa_uncached_ptr((void *)&static_tag),
+	zassert_equal(&static_tag, sys_cache_uncached_ptr_get((void *)&static_tag),
 			  "stack memory not cached");
 
 	/* Un/cached regions should be working */
 	printk(" Cache behavior check\n");
-	volatile int *ctag = (volatile int *)arch_xtensa_cached_ptr((void *)&tag);
-	volatile int *utag = (volatile int *)arch_xtensa_uncached_ptr((void *)&tag);
+	volatile int *ctag = (volatile int *)sys_cache_cached_ptr_get((void *)&tag);
+	volatile int *utag = (volatile int *)sys_cache_uncached_ptr_get((void *)&tag);
```

Comment:
all "uncached get" have a cached pointer as their argument, so the latter one should have `(void __sparce_cache *)` in the argument conversion